### ISSUE TITLE
osTicket LDAP Plugin - Connection Information Issue

### ISSUE THREAD

**Issue by cxiong84** [osTicket/osTicket-plugins#163]
Trying to setup LDAP search with most recent version of osTicket 1.12.2 and LDAP Plugin, but getting this error:
**Content cannot start with the following characters: = - + @** in the password field.
Yes the ldap user password does start with a character. Can this be fixed so this will work regardless of what the password starts with? This LDAP account is being used in other places so No I don't want to change it and cause all kinds of problems.

**Comment by cxiong84** [comment]
Bump. No updates?

**Comment by JediKev** [comment]
@cxiong84 

This is to prevent Formula Injections via CSV (security vulnerability for mostly Microsoft products). It applied to any Save-able field but I think we should ignore formula injection attempts via password fields because you cannot export passwords. We will look into this.

Cheers.

**Comment by ntozier** [comment]
Rather than ignore it... maybe just make a mechanism for "this is a password" check it for injection attacks... but save it in its correct format anyway.

**Comment by JediKev** [comment]
@ntozier 

Anything starting with `= - + @` is considered a Formula Injection 😉 So we are checking it for that currently.

Cheers.

**Comment by cxiong84** [comment]
So I was able to find a way around this. Made an edit to the class.validator.php file in the Include folder. Removed the @ from the following line:
    static function is_formula($text, &$error='') {
        **if (!preg_match('/^[^=\+-@].*$/s', $text))**
            $error = __('Content cannot start with the following characters: = - + @');
        return $error == '';

So it wasn't in the plugin, it was in the osticket files itself.

**Comment by JediKev** [comment]
@cxiong84 

I hope you know that doing so opens you up to Formula Injections in the entire software. This is not a good idea. I'm currently working on a fix now so stay tuned.

Cheers.

**Comment by cxiong84** [comment]
Yep understood but this is internal and not available for external use. It's only in a test environment at this time as well.

**Comment by JediKev** [comment]
@cxiong84 

So, below is a new build of the plugin with the fix applied:
[auth-ldap.phar.zip](https://github.com/osTicket/osTicket-plugins/files/4300171/auth-ldap.phar.zip)

You will also need to apply these changes:
- https://github.com/osTicket/osTicket/pull/5398

Once you make the changes and replace the phar, you should be good to go without having to disable Formula Injection prevention.

Cheers.

**Comment by cxiong84** [comment]
It looks like it's partially working now. However, I had the DNS Servers field populated with an IP address, it had the same error in that field. If I leave the field blank, it seems to work just fine and I can save the LDAP settings.

**Comment by JediKev** [comment]
@cxiong84 

Can you post a screenshot of what you mean or post an example of the IPs you used for the DNS Servers field? IPs shouldn't start with `= - + @` so I don't know why the error occurred.

Cheers.

**Comment by cxiong84** [comment]
This is what it shows when it errors out on that field after clicking to save changes.
![image](https://user-images.githubusercontent.com/55559541/76252710-433c0c00-6217-11ea-85b6-8d474d615fa1.png)

**Comment by JediKev** [comment]
@cxiong84 

I'm not seeing the same issues. With the patch in place, and using the new phar from above, it saves successfully with IP addresses in the field:
![Screen Shot 2020-03-09 at 15 06 40](https://user-images.githubusercontent.com/11823401/76253031-e68d2100-6217-11ea-99db-73159b501a47.png)

I'm not sure what your issue could be..? As you can see from your screenshot, the content does not start with any of those characters. Try looking at the POST data to see if maybe after posting something adds one of the characters to the beginning.

Cheers.

**Comment by cxiong84** [comment]
Thanks for checking. I'm still running the 1.12.2 of from the original post. I'll try upgrading to see if that helps. Otherwise, I can do without the DNS servers.

**Comment by JediKev** [comment]
@cxiong84 

After you upgrade, don't forget to apply osTicket/osTicket#5398 again.

Cheers.
